frontend/js/src/common/brainzplayer/InternetArchivePlayer.tsx

```diff
@@ -0,0 +1,197 @@
+import React from "react";
+import { faArchive } from "@fortawesome/free-solid-svg-icons";
+import { DataSourceProps, DataSourceType } from "./BrainzPlayer";
+import { getTrackName, getArtistName } from "../../utils/utils";
+
+type IARecording = {
+  track_id: string;
+  name: string;
+  artist: string[];
+  album?: string;
+  stream_urls: string[];
+  artwork_url?: string;
+};
+
+type State = {
+  loading: boolean;
+  currentTrack: IARecording | null;
+};
+
+export default class InternetArchivePlayer
+  extends React.Component<DataSourceProps, State>
+  implements DataSourceType {
+  public name = "internetArchive";
+  public domainName = "archive.org";
+  public icon = faArchive;
+  public iconColor = "#6c757d";
+  audioRef: React.RefObject<HTMLAudioElement>;
+
+  constructor(props: DataSourceProps) {
+    super(props);
+    this.audioRef = React.createRef();
+    this.state = {
+      loading: false,
+      currentTrack: null,
+    };
+  }
+
+  handleAudioEnded = () => {
+    const { onTrackEnd } = this.props;
+    onTrackEnd();
+  };
+
+  handleTimeUpdate = () => {
+    const { onProgressChange } = this.props;
+    if (this.audioRef.current) {
+      onProgressChange(this.audioRef.current.currentTime * 1000);
+    }
+  };
+
+  handleLoadedMetadata = () => {
+    const { onDurationChange } = this.props;
+    if (this.audioRef.current) {
+      onDurationChange(this.audioRef.current.duration * 1000);
+    }
+  };
+
+  searchAndPlayTrack = async (listen: any) => {
+    const { onTrackNotFound, handleError } = this.props;
+    const trackName = getTrackName(listen);
+    const artistName = getArtistName(listen);
+
+    if (!trackName && !artistName) {
+      onTrackNotFound();
```

Comment:
Let's warn the user like we do in other music players:

https://github.com/metabrainz/listenbrainz-server/blob/a6f6f8e092c5b7ee7ce496db901d7ec577acba7e/frontend/js/src/common/brainzplayer/SpotifyPlayer.tsx#L190-L194